### PR TITLE
python3Packages.diagrams: init at 0.12.0

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

Adding popular diagrams package to draw cloud diagrams with python. In addition this PR upgrades the over 2 years old graphviz package.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/graphviz/hardcode-graphviz-path.patch [hunk 1]
```diff
@@ -67,26 +59,18 @@ index 7ec12f7..2e8550d 100644
...
 -    Popen.assert_called_once_with(['dot', '-V'],
 +    Popen.assert_called_once_with(['@graphviz@/bin/dot', '-V'],
                                    stdout=subprocess.PIPE,
                                    stderr=subprocess.STDOUT,
                                    startupinfo=mocker.ANY)
-@@ -211,7 +214,7 @@ def test_version_mocked(mocker, Popen):
+@@ -269,7 +270,7 @@ def test_version_mocked(mocker, Popen, stdout, expected):  # noqa: N803
  
-     assert version() == (1, 2, 3)
+     assert version() == expected
```
Comment (jonringer) [inline pkgs/development/python-modules/graphviz/hardcode-graphviz-path.patch:73]: some of these patch changes seem like they should be upstreamed, did you want to open a PR against them and then just use `fetchpatch` to bring in those changes?

The nix-specific changes should stay in nixpkgs though.

File: pkgs/development/python-modules/diagrams/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    graphviz
+    jinja2
+  ];
+
+  # add image resources to build
+  postInstall = ''
+    cp -r resources $out/${python.sitePackages}/
+  '';
+
+  # tests not included with PyPi Release
```
Comment (dotlambda) [inline pkgs/development/python-modules/diagrams/default.nix:36]: Can you ask the project to distribute them?

File: pkgs/development/python-modules/graphviz/hardcode-graphviz-path.patch (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/graphviz/hardcode-graphviz-path.patch:73]: This patch has only nix-specific changes.

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**